### PR TITLE
Avoid the copy of a large 4D array if possible

### PR DESCRIPTION
**Pull request by mathrack**

### CODE REVIEW

File: src/case.f90 [hunk 1]
```diff
@@ -233,58 +233,73 @@ end subroutine preprocessing
...
+      call real_postprocessing(rho1, ux1, uy1, uz1, pp3, T, ep1)
     else
-      T = phi1
+      call real_postprocessing(rho1, ux1, uy1, uz1, pp3, phi1, ep1)
     endif
 
+  end subroutine postprocessing
+  !##################################################################
+  !##################################################################
+  subroutine real_postprocessing(rho1, ux1, uy1, uz1, pp3, phi1, ep1)
```
Comment (pbartholomew08) [inline src/case.f90]: I'd suggest "run" postprocessing (or something like that), "real" could be misinterpreted as being related to the data type.

Comment (mathrack) [reply src/case.f90]: Fixed in https://github.com/xcompact3d/Incompact3d/pull/176/commits/066394b1a5a99f13bb5eabfd58d8a85579a6722f

**Review by pbartholomew08 — APPROVED** [review]

**Comment by mathrack** [comment]
@pbartholomew08 shall we merge this PR or update it to avoid a large array on the stack ? I could move
```
real(mytype),dimension(xsize(1),xsize(2),xsize(3),numscalar) :: T ! FIXME This can be huge
```
to the ABL module and allocate it at the beginning of the simulation ?

**Comment by pbartholomew08** [comment]
> @pbartholomew08 shall we merge this PR or update it to avoid a large array on the stack ? I could move
> 
> ```
> real(mytype),dimension(xsize(1),xsize(2),xsize(3),numscalar) :: T ! FIXME This can be huge
> ```
> 
> to the ABL module and allocate it at the beginning of the simulation ?

How about we make it allocatable and allocate it only in the ABL case? 

I don't really understand why it doesn't just use the normal scalar variables for this and something like ```scalar_index_T``` to identify which scalar to use, however I think this should be a separate work as getting rid of the memory usage in the general case is immediately useful.

**Comment by mathrack** [comment]
> 
> How about we make it allocatable and allocate it only in the ABL case?
> 
> I don't really understand why it doesn't just use the normal scalar variables for this and something like `scalar_index_T` to identify which scalar to use, however I think this should be a separate work as getting rid of the memory usage in the general case is immediately useful.

I will merge this and open another PR to fix this ABL-specific stuff

**Pull request merged**